src/authentication.rs

```diff
@@ -0,0 +1,158 @@
+use nex_rs::client::{ClientConnection};
+use nex_rs::nex_types::StructureInterface;
+use nex_rs::packet::{Packet, PacketV1};
+use nex_rs::stream::{StreamIn};
+use no_std_io::{Cursor, Reader, StreamReader};
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+
+pub const AUTHENTICATION_PROTOCOL_ID: u8 = 0xA;
+
+#[derive(Debug, Clone, Copy, PartialEq, TryFromPrimitive, IntoPrimitive)]
+#[repr(u32)]
+pub enum AuthenticationMethod {
+    Login = 0x1,
+    LoginEx = 0x2,
+    RequestTicket = 0x3,
+    GetPID = 0x4,
+    GetName = 0x5,
+    LoginWithParam = 0x6,
+}
+
+#[derive(Default)]
+pub struct NintendoLoginData(String);
```

Comment:
It doesn't look like this is currently used.  Why is it here?

Reply:
Used from the port, I believe it's a structure used in the user-defined functions but honestly can be removed.